graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,331 @@
+import numpy as np
+from ..match import GraphMatch as GMP
+from sklearn.base import BaseEstimator
+import itertools
+
+
+class VNviaSGM(BaseEstimator):
+    """
+    This class implements Vertex Nomination via Seeded Graph Matching
+    (VNviaSGM) with the algorithm described in [1].
+
+    VNviaSGM is a nomination algorithm, so instead of completely matching two
+    graphs `A` and `B`, it proposes a nomination list of potential matches in
+    graph `B` to a vertex of interest (voi) in graph `A` with associated
+    probabilities. VNviaSGM generates an initial induced subgraph about the
+    voi to determine which seeds are close enough to be used.
+    All the seeds that are close enough are then used to generate subgraphs in
+    both A and B. These subgraphs are matched using the seeded graph matching
+    algorithm (SGM), and a nomination list is returned.
+
+    Parameters
+    ----------
+    order_voi_subgraph: int, positive (default = 1)
+        Order used to create induced subgraph on `A` about voi where the max
+        distance between voi and other nodes is order_voi_subgraph.
+        This induced subgraph will be used to determine what seeds are used
+        when the SGM algorithm is called. If no seeds are in this subgraph
+        about voi, then VNviaSGM will return None.
+
+    order_seeds_subgraph: int, positive (default = 1)
+        Order used to  create induced subgraphs on `A`
+        and `B`. These subgraphs are centered about the seeds that were determined
+        by the subgraph generated by `order_voi_subgraph`. These two subgraphs will
+        be passed into the SGM algorithm.
+
+    n_init: int, positive (default = 100)
+        Number of restarts for soft seeded graph matching algorithm. Increasing the
+        number of restarts will make the probabilites returned more precise.
+
+    Attributes
+    ----------
+    n_seeds: int
+        Number of seeds passed in `seedsA` that occured in the induced subgraph about `voi`
+
+    nomination_list: 2d-array
```

Comment:
how large is the nomination list? Does it sort all vertices not in seeds? If so I feel like there should be a parameter capping the number

Reply:
The nomination list only returns the vertices that are in graph B in the induced subgraph with max distance order_seeds_subgraph centered around the seeds close to voi. It is usually not all of the nodes in the graph; however, for small graphs or highly connected graphs it may be. I will add this as an optional argument.